Objective-C/CBLReplicator.mm

```diff
@@ -505,14 +541,7 @@ - (bool) handleError: (C4Error)c4err {
         auto delay = retryDelay(++_retryCount);
         CBLLogInfo(Sync, @"%@: Transient error (%@); will retry in %.0f sec...",
                    self, error.localizedDescription, delay);
-        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delay * NSEC_PER_SEC)),
-                       _dispatchQueue, ^{
-                           CBL_LOCK(self) {
-                               // Retry if still in offline state:
-                               if (_state == kCBLStateOffline)
-                                   [self retry: NO];
-                           }
-                       });
+        [self scheduleRetry: delay];
```

Comment:
Also we don't want the retry to be executed if the replicator is suspended.

Reply:
Another way to implement this is to cancel the previous retry before suspending so we don't need to check here.